src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVFileUtils.java

```diff
@@ -0,0 +1,176 @@
+package org.broadinstitute.hellbender.tools.spark.sv.utils;
+
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchSet;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.*;
+import java.util.*;
+
+public final class SVFileUtils {
+
+    private static final String REFERENCE_GAP_INTERVAL_FILE_COMMENT_LINE_PROMPT = "#";
+
+    /**
+     * Creates a directory, in local FS, HDFS, or Google buckets to write individual files in.
+     */
+    public static void createDirectory(final String pathString) throws IOException {
+        Utils.nonNull(pathString);
+
+        if (BucketUtils.isHadoopUrl(pathString)) {
+            final boolean isSuccessful = org.apache.hadoop.fs.FileSystem.get(new org.apache.hadoop.conf.Configuration())
+                            .mkdirs(new org.apache.hadoop.fs.Path(pathString));
+            if (!isSuccessful)
+                throw new IOException("Dir creation failed: " + pathString);
+        } else {
+            final java.nio.file.Path dir = java.nio.file.Files.createDirectory(IOUtils.getPath(pathString)); // createDirectory checks and throws if a file with same name exists
+            final boolean isSuccessful = java.nio.file.Files.isDirectory(dir) && java.nio.file.Files.isWritable(dir);
+            if (!isSuccessful)
+                throw new IOException("File created is either not a directory or not writable: " + pathString);
+        }
+    }
+
+    public static void writeSAMFile(final String outputName, final Iterator<SAMRecord> alignments, final SAMFileHeader header,
+                                    final boolean preOrdered) {
+        Utils.nonNull(alignments, "provided alignments to write out is null");
+        Utils.nonNull(header, "provided header for outputting sam file is null");
+        Utils.nonNull(outputName, "provided output name is null");
+
+        try (final SAMFileWriter writer = getSAMFileWriter(outputName, header, preOrdered)) {
+            alignments.forEachRemaining(writer::addAlignment);
+        } catch ( final UncheckedIOException ie) {
+            throw new GATKException("Can't write SAM file to the specified location: " + outputName, ie);
+        }
+    }
+
+    private static SAMFileWriter getSAMFileWriter(final String outputName, final SAMFileHeader header, final boolean preOrdered) {
```

Comment:
The `ReadUtils` class will have support for HDFS and other `java.nio.Path` representation here: https://github.com/broadinstitute/gatk/pull/2558